repos/system_upgrade/common/actors/checkyumpluginsenabled/libraries/checkyumpluginsenabled.py

```diff
@@ -0,0 +1,72 @@
+import os
+
+from leapp import reporting
+from leapp.libraries.common.config.version import get_source_major_version
+from leapp.libraries.common.rhsm import skip_rhsm
+
+# If LEAPP_NO_RHSM is set, subscription-manager and product-id will not be
+# considered as required when checking whether the required plugins are enabled.
+REQUIRED_YUM_PLUGINS = {'subscription-manager', 'product-id'}
+FMT_LIST_SEPARATOR = '\n    - '
+
+
+def check_required_yum_plugins_enabled(yum_config):
+    """
+    Checks whether the yum plugins required by the IPU are enabled.
+
+    If they are not enabled, a report is produced informing the user about it.
+
+    :param yum_config: YumConfig
+    """
+
+    missing_required_plugins = REQUIRED_YUM_PLUGINS - set(yum_config.enabled_plugins)
+
+    if skip_rhsm():
+        missing_required_plugins -= {'subscription-manager', 'product-id'}
+
+    if missing_required_plugins:
+        missing_required_plugins_text = ''
+        for missing_plugin in missing_required_plugins:
+            missing_required_plugins_text += '{0}{1}'.format(FMT_LIST_SEPARATOR, missing_plugin)
+
+        if get_source_major_version() == '7':
+            pkg_manager = 'YUM'
+            pkg_manager_config_path = '/etc/yum.conf'
+            plugin_configs_dir = '/etc/yum/pluginconf.d'
+        else:
+            # On RHEL8+ the yum package might not be installed
+            pkg_manager = 'DNF'
+            pkg_manager_config_path = '/etc/dnf/dnf.conf'
+            plugin_configs_dir = '/etc/dnf/plugins'
+
+        # pkg_manager_config_path(plugins=1) - Allows YUM to load plugins
+        # subscription_manager_plugin_conf, product_id_plugin_conf - Plugins can be enabled plugins individually
```

Comment:
Hmm this comment is a bit cryptic imho, especially the 'Plugins can be enabled plugins'. Could you please rephrase that?

Reply:
Thanks for catching this, the comment was indeed cryptic. I rephrased this (latest push).